src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c

```diff
@@ -1258,9 +1258,11 @@ static int xclmgmt_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 
 	/*
 	 * Even if extended probe fails, make sure feature ROM subdev
-	 * is loaded to provide basic info about the board.
+	 * is loaded to provide basic info about the board. Also, need
+	 * FLASH to be able to flash new shell.
 	 */
 	(void) xocl_subdev_create_by_id(lro, XOCL_SUBDEV_FEATURE_ROM);
+	(void) xocl_subdev_create_by_id(lro, XOCL_SUBDEV_FLASH);
```

Comment:
Why the flash did not create by following
	(void) xocl_subdev_create_by_level(lro, XOCL_SUBDEV_LEVEL_BLD);
	(void) xocl_subdev_create_vsec_devs(lro);

Reply:
this is on a non-ssv2 platform, so you don't have blp/vsec, right?